### PR TITLE
Add cross-encoder model documentation

### LINKED ISSUE #6352: [DOC] Add cross-encoder model
**What do you want to do?**

We will release a new model type in 2.12 https://github.com/opensearch-project/ml-commons/pull/1615. Need to update doc for adding this model. 

- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Closes #6352 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/custom-local-models.md [hunk 1]
```diff
@@ -315,4 +315,149 @@ The response contains the tokens and weights:
...
 ## Step 5: Use the model for search
 
-To learn how to use the model for vector search, see [Set up neural search]({{site.url}}{{site.baseurl}}http://localhost:4000/docs/latest/search-plugins/neural-search/#set-up-neural-search).
+To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).
+
+## Cross-encoder models
+
+Cross-encoder models support query reranking. 
+
+To register a cross-encoder model, send a request in the following format. The `model_config` object is optinoal. Cross-encoder models' `function_name` is `TEXT_SIMILARITY`. For example, the following request registers a `ms-marco-TinyBERT-L-2-v2` model:
```
Comment (HenryL27) [inline _ml-commons-plugin/custom-local-models.md:324]: optinoal -> optional

File: _search-plugins/search-relevance/reranking-search-results.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Introduced 2.12
 You can rerank search results using a cross-encoder reranker in order to improve search relevance. To implement reranking, you need to configure a [search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/) that runs at search time. The search pipeline intercepts search results and applies the [`rerank` processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rerank-processor/) to them. The `rerank` processor evaluates the search results and sorts them based on the new scores provided by the cross-encoder model. 
 
 **PREREQUISITE**<br>
-Before using hybrid search, you must set up a cross-encoder model. For more information, see [Choosing a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#choosing-a-model).
+Before using hybrid search, you must set up a cross-encoder model. For more information, see [Cross-encoder models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/custom-local-models/#cross-encoder-models).
```
Comment (HenryL27) [inline _search-plugins/search-relevance/reranking-search-results.md:16]: Prerequisite for reranking, not hybrid search.

**Review by HenryL27 — APPROVED** [review]
Couple tweaks, but otherwise lgtm!

File: _search-plugins/search-relevance/reranking-search-results.md (hunk 2 above)
Comment (kolchfa-aws) [reply _search-plugins/search-relevance/reranking-search-results.md:16]: Thank you! Good catch.

Comment (kolchfa-aws) [inline _search-plugins/search-relevance/reranking-search-results.md:16]:
```suggestion
Before configuring a reranking pipeline, you must set up a cross-encoder model. For more information, see [Cross-encoder models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/custom-local-models/#cross-encoder-models).
```

File: _ml-commons-plugin/custom-local-models.md (hunk 1 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/custom-local-models.md:324]:
```suggestion
To register a cross-encoder model, send a request in the following format. The `model_config` object is optional. Cross-encoder models' `function_name` is `TEXT_SIMILARITY`. For example, the following request registers a `ms-marco-TinyBERT-L-2-v2` model:
```

**Review by vagimeli — APPROVED** [review]
LGTM, with minimal edits.

File: _ml-commons-plugin/custom-local-models.md [hunk 3]
```diff
@@ -315,4 +315,149 @@ The response contains the tokens and weights:
...
+    "description": "test model",
+    "model_format": "TORCH_SCRIPT",
+    "model_group_id": "lN4AP40BKolAMNtR4KJ5",
+    "model_content_hash_value": "90e39a926101d1a4e542aade0794319404689b12acfd5d7e65c03d91c668b5cf",
+    "model_config": { 
+        "model_type": "bert",
+        "embedding_dimension": 1,
+        "framework_type": "huggingface_transformers",
+        "total_chunks":2,
+        "all_config": "{\"total_chunks\":2,\"is_hidden\":false}"
```
[the remark this answers is not in the document]
Comment (dhrubo-os) [reply _ml-commons-plugin/custom-local-models.md:341]: otherwise looks good.

**Review by dhrubo-os — APPROVED** [review]

File: _ml-commons-plugin/custom-local-models.md [hunk 4]
```diff
@@ -315,4 +315,149 @@ The response contains the tokens and weights:
...
 ## Step 5: Use the model for search
 
-To learn how to use the model for vector search, see [Set up neural search]({{site.url}}{{site.baseurl}}http://localhost:4000/docs/latest/search-plugins/neural-search/#set-up-neural-search).
+To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).
+
+## Cross-encoder models
+
+Cross-encoder models support query reranking. 
+
+To register a cross-encoder model, send a request in the following format. The `model_config` object is optional. For cross-encoder models, specify the `function_name` as `TEXT_SIMILARITY`. For example, the following request registers a `ms-marco-TinyBERT-L-2-v2` model:
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:324]:
```suggestion
To register a cross-encoder model, send a request in the following format. The `model_config` object is optional. For cross-encoder models, specify the `function_name` as `TEXT_SIMILARITY`. For example, the following request registers an `ms-marco-TinyBERT-L-2-v2` model:
```

File: _ml-commons-plugin/custom-local-models.md [hunk 5]
```diff
@@ -315,4 +315,149 @@ The response contains the tokens and weights:
...
+            "order": "LITTLE_ENDIAN"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+Higher document score means higher similarity. In the preceding response, documents are scored as follows against the query text `today is sunny`:
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:452]: Either "A higher document score" or "Higher document scores"

File: _ml-commons-plugin/pretrained-models.md [hunk 6]
```diff
@@ -296,4 +296,4 @@ The following table provides a list of sparse encoding models and artifact links
...
\ No newline at end of file
+| `amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1` | 1.0.1 | Yes | - [model_url](https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1/1.0.1/torch_script/neural-sparse_opensearch-neural-sparse-tokenizer-v1-1.0.1-torch_script.zip)<br>- [config_url](https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1/1.0.1/torch_script/config.json) | A neural sparse tokenizer model. The model tokenizes text into tokens and assigns each token a predefined weight, which is the token's IDF (if the IDF file is not provided, the weight defaults to 1). For more information, see [Preparing a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/custom-local-models/#preparing-a-model). |
```
Comment (natebower) [inline _ml-commons-plugin/pretrained-models.md:299]: Please define IDF.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just a few changes. Thanks!

File: _ml-commons-plugin/custom-local-models.md (hunk 5 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/custom-local-models.md:452]:
```suggestion
A higher document score means higher similarity. In the preceding response, documents are scored as follows against the query text `today is sunny`:
```

File: _ml-commons-plugin/pretrained-models.md (hunk 6 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/pretrained-models.md:299]:
```suggestion
| `amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1` | 1.0.1 | Yes | - [model_url](https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1/1.0.1/torch_script/neural-sparse_opensearch-neural-sparse-tokenizer-v1-1.0.1-torch_script.zip)<br>- [config_url](https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1/1.0.1/torch_script/config.json) | A neural sparse tokenizer model. The model tokenizes text into tokens and assigns each token a predefined weight, which is the token's inverse document frequency (IDF). If the IDF file is not provided, the weight defaults to 1. For more information, see [Preparing a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/custom-local-models/#preparing-a-model). |
```

**Pull request merged**